sp/src/game/client/mapbase/vscript_vgui.cpp

```diff
@@ -0,0 +1,1541 @@
+//========= Mapbase - https://github.com/mapbase-source/source-sdk-2013 ============//
+//
+// Purpose:
+//
+// $NoKeywords: $
+//=============================================================================//
+
+
+#include "cbase.h"
+#include "tier1/utlcommon.h"
+
+#include "inputsystem/iinputsystem.h"
+
+#include <vgui/VGUI.h>
+#include <vgui/IVGui.h>
+#include <vgui/ISurface.h>
+#include <vgui/IInput.h>
+#include <vgui/ILocalize.h>
+
+#include <vgui_controls/Controls.h>
+#include <vgui_controls/Panel.h>
+#include <vgui_controls/Label.h>
+//#include <vgui_controls/Button.h>
+
+#include <ienginevgui.h>
+
+#include "view.h"
+
+#include "vscript_vgui.h"
+#include "vscript_vgui.nut"
+
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+
+//
+//
+// Exposing a new panel class (e.g. vgui::FileOpenDialog):
+//
+// Create 'class IScript_FileOpenDialog : IScript_Frame' with C++ bindings.
+// Create 'class CScript_FileOpenDialog : FileOpenDialog' with vgui message callbacks and overrides (if needed)
+// Create 'class CScript_FileOpenDialog_Helper : IScript_FileOpenDialog< CScript_FileOpenDialog >' for the script bindings
+// Add new condition in CScriptVGUI::Create
+//
+//
+
+
+using namespace vgui;
+class IScriptVGUIObject;
+class IScriptVGUIPanel;
+
+static inline unsigned int IntToHandle( int i )
+{
+	if ( i < 0 )
+		return 0;
+	return static_cast<unsigned int>(i);
+}
+
+static inline int HandleToInt( unsigned int i )
+{
+	Assert( i <= INT_MAX );
```

Comment:
These handles are vector indices, and are in tens or hundreds. I'm not sure what would happen if it were to exceed INT_MAX in a malicious attempt because CUtlVector works with signed int.

This check here is not necessary because internally the returned values are signed. The assert is just redundancy.